doc/rule-dsl.md

```diff
@@ -0,0 +1,510 @@
+* `udp`
+
+#### Property: match.source <a id="match-source"></a>
+
+The `source` field is used to qualify a rule to only apply to requests from a specific caller.
+For example,
+the following `match` clause qualifies the rule to only apply to calls from the "reviews" microservice.
+
+```yaml
+destination: ratings.default.svc.cluster.local
+match:
+  source: reviews.default.svc.cluster.local
+```
+
+#### Property: match.source_tags <a id="match-source_tags"></a>
+
+The `source_tags` field can be used to further qualify a rule to only apply to specific instances of a
+calling service.
+For example,
+the following `match` clause refines the previous example to qualify the rule to only 
+apply to calls from version "v2" of the "reviews" microservice.
+
+```yaml
+destination: ratings.default.svc.cluster.local
+match:
+  source: reviews.default.svc.cluster.local
+  source_tags:
+    version: v2
+```
+
+#### Property: match.http <a id="match-http"></a>
+
+The `http` field is a set of one or more property-value pairs where each property is an HTTP header name
+and the corresponding value is one of the following:
+
+1. `exact: value`, where the header value must match `value` exactly
+2. `prefix: value`, where `value` must be a prefix of the header value
+3. `regex: value`, where `value` is a regular expression that matches the header value
+
+For example, the following rule will only apply
+to an incoming request if it includes a "Cookie" header that contains the substring "user=jason".
+
+```yaml
+destination: reviews.default.svc.cluster.local
+match:
+  http:
+    Cookie:
+      regex: "^(.*?;)?(user=jason)(;.*)?$"
+```
+
+If more than one property-value pair is provided, then all of the corresponding headers
+must match for the rule to apply.
+
+The `source` and `http` fields can both be set in the `match` object in which case both criteria must pass.
+For example, the following rule only applies if the source of the request is "reviews:v2" 
+AND the "Cookie" header containing "user=jason" is present. 
+
+```yaml
+destination: ratings.default.svc.cluster.local
+match:
+  source: reviews.default.svc.cluster.local
+  source_tags:
+    version: v2
+  http:
+    Cookie:
+      regex: "^(.*?;)?(user=jason)(;.*)?$"
+```
+
+#### Property: match.tcp <a id="match-tcp"></a>
+
+TBD
+
+#### Property: match.udp <a id="match-udp"></a>
+
+TBD
+
+#### Property: route <a id="route"></a>
+
+The `route` field identifies a set of one or more the weighted backends to call when the rule is activated.
+Each `route` backend is an object with the following fields:
+
+* `tags`
+* `weight`
+* `destination`
+
+Of these fields, `tags` is the only required one, the others are optional.
+
+#### Property: route.tags <a id="route-tags"></a>
+
+The `tags` field is a list of instance tags to identify the target instances (e.g., version) to route requests to.
+If there are multiple registered instances with the specified tag(s), they will be routed to in a round-robin fashion.
+
```

Comment:
requests will be load balanced among these instances based on the load balancing policy set in the Destination Policy